### ISSUE TITLE
text/x-handlebars MIME Type not supported in sprocket

### ISSUE THREAD

**Issue by imechemi** [rails/sprockets#260]
Hi, after upgrading rails from **4.2.4** to **4.2.5.1**, the sprocket gem also got updated from 3.4.0 to 3.5.2. And my application broke when it has to load .hamlbars file from assets.

The backtrace from app ends at this [line](https://github.com/rails/sprockets/blob/v3.5.2/lib/sprockets/legacy.rb#L111) from sprocket when the `text/x-handlebars` mime type is not found.

Below are the MIME types I found using `puts` from sprockets at that [line](https://github.com/rails/sprockets/blob/v3.5.2/lib/sprockets/legacy.rb#L111) and it doesn't have `"text/x-handlebars"` in the list.

```
{"application/javascript"=>{:extensions=>[".js"], :charset=>#<Method: Module(Sprockets::EncodingUtils)#detect_unicode>}, 
"application/json"=>{:extensions=>[".json"], :charset=>#<Method: Module(Sprockets::EncodingUtils)#detect_unicode>}, 
"application/xml"=>{:extensions=>[".xml"]}, 
"text/css"=>{:extensions=>[".css"], :charset=>#<Method: Module(Sprockets::EncodingUtils)#detect_css>}, 
"text/html"=>{:extensions=>[".html", ".htm"], :charset=>#<Method: Module(Sprockets::EncodingUtils)#detect_html>}, 
"text/plain"=>{:extensions=>[".txt", ".text"], :charset=>#<Method: Module(Sprockets::EncodingUtils)#detect>}, 
"text/yaml"=>{:extensions=>[".yml", ".yaml"], :charset=>#<Method: Module(Sprockets::EncodingUtils)#detect_unicode>}, 
"image/x-icon"=>{:extensions=>[".ico"]}, 
"image/bmp"=>{:extensions=>[".bmp"]}, 
"image/gif"=>{:extensions=>[".gif"]}, 
"image/webp"=>{:extensions=>[".webp"]}, 
"image/png"=>{:extensions=>[".png"]}, 
"image/jpeg"=>{:extensions=>[".jpg", ".jpeg"]}, 
"image/tiff"=>{:extensions=>[".tiff", ".tif"]}, 
"image/svg+xml"=>{:extensions=>[".svg"]}, 
"video/webm"=>{:extensions=>[".webm"]},
"audio/basic"=>{:extensions=>[".snd", ".au"]}, 
"audio/aiff"=>{:extensions=>[".aiff"]}, 
"audio/mpeg"=>{:extensions=>[".mp3", ".mp2", ".m2a", ".m3a"]}, 
"application/ogg"=>{:extensions=>[".ogx"]}, 
"audio/midi"=>{:extensions=>[".midi", ".mid"]}, 
"video/avi"=>{:extensions=>[".avi"]}, 
"audio/wave"=>{:extensions=>[".wav", ".wave"]}, 
"video/mp4"=>{:extensions=>[".mp4", ".m4v"]}, 
"application/vnd.ms-fontobject"=>{:extensions=>[".eot"]}, 
"application/x-font-ttf"=>{:extensions=>[".ttf"]}, 
"application/font-woff"=>{:extensions=>[".woff"]}}
```

I am not sure if this is handlebars_assets or sprocket issue. If this is not related to sprocket, I would like to know what is causing this and I will be grateful if someone can provide some context before closing this issue. thanks.

**Comment by schneems** [comment]
If you downgrade you sprockets version does it work again?

**Comment by imechemi** [comment]
Yes instead I downgraded sprocket-rails to 2.2.3 and it worked. It was not sprocket issue I think.

**Comment by schneems** [comment]
Can you try with the latest version of sprockets-rails 3.0.4? If that doesn't work can you open an issue there? It would help to have a small app that reproduces the problem.

**Comment by rafaelfranca** [comment]
This seems like `handlebars_assets` incompatibility with sprockets-rails 3.

**Comment by schneems** [comment]
Do we know what it is? Would it help to add deprecation warnings in sprockets 2?

**Comment by rafaelfranca** [comment]
I don't think so, we didn't changed the mime-type API so it seems just like a code bug in handlebars_assets or maybe the applications is just using a wrong version of the gem.

**Comment by imechemi** [comment]
> This seems like handlebars_assets incompatibility with sprockets-rails 3.

@rafaelfranca Sorry for my late reply, I am using an older version of handlebar_assets-0.14.0 in my Gemfile and I cannot update that since I have a dependency locked from another gem. May be If I could update the handlerbar_assets, It might work with latest version of sprocket-rails.

So for now I have kept my sprocket-rails version to 2.2.3. And that didn't break anything. 


> Do we know what it is? Would it help to add deprecation warnings in sprockets 2?

@schneems  It would be nice if there is deprecation warnings, I had to dig out entire stack trace and see where is this is landing.
